File: tests/machine/x/fortran/nested_function.f90
program nested_function
  implicit none
  print *, outer(3)
contains
  function outer(x) result(res)
    integer, intent(in) :: x
    integer :: res
    res = inner(x, 5)
  end function outer

  function inner(x, y) result(res_inner)
    integer, intent(in) :: x
    integer, intent(in) :: y
    integer :: res_inner
    res_inner = x + y
  end function inner
end program nested_function

File: tests/human/x/pl/nested_function.pl
:- style_check(-singleton).
        outer(X, Res) :-
            catch(
                (
                inner(Y, Res) :-
                _V0 is X + Y,
                Res = _V0.
                    ,
                    true
                )
                , return(_V1),
                    Res = _V1
                )
            .
            outer(X, Res) :-
            inner(5, _V2),
            Res = _V2.

    main :-
    outer(3, _V3),
    write(_V3),
    nl
    .
:- initialization(main, main).
